engine/src/main/java/org/pentaho/di/trans/steps/csvinput/CsvInput.java

```diff
@@ -0,0 +1,1179 @@
+                  data.moveEndBufferPointerXTimes( data.encodingType.getLength() );
+                } else {
+                  //Re-check for a new Line
+                  data.moveEndBufferPointerXTimes( data.encodingType.getLength() );
+                  if ( !data.newLineFound() ) {
+                    throw new KettleFileException( BaseMessages.getString( PKG, "TextFileInput.Log.SingleLineFound" ) );
+                  }
+                }
+              }
+              break;
+            case TextFileInputMeta.FILE_FORMAT_MIXED:
+              if ( data.isCarriageReturn() || doubleLineEnd ) {
+                data.moveEndBufferPointerXTimes( data.encodingType.getLength() );
+              }
+              break;
+          }
+        }
+
+        data.setStartBuffer( data.getEndBuffer() );
+      }
+
+      // See if we reached the end of the line.
+      // If not, we need to skip the remaining items on the line until the next newline...
+      //
+      if ( !newLineFound && !data.resizeBufferIfNeeded() ) {
+        do {
+          data.moveEndBufferPointer();
+          if ( data.resizeBufferIfNeeded() ) {
+            break; // nothing more to read.
+          }
+
+          // TODO: if we're using quoting we might be dealing with a very dirty file with quoted newlines in trailing
+          // fields. (imagine that)
+          // In that particular case we want to use the same logic we use above (refactored a bit) to skip these fields.
+
+        } while ( !data.newLineFound() );
+
+        if ( !data.resizeBufferIfNeeded() ) {
+          while ( data.newLineFound() ) {
+            data.moveEndBufferPointer();
+            if ( data.resizeBufferIfNeeded() ) {
+              break; // nothing more to read.
+            }
+          }
+        }
+
+        // Make sure we start at the right position the next time around.
+        data.setStartBuffer( data.getEndBuffer() );
+      }
+
+      // Optionally add the current filename to the mix as well...
+      //
+      if ( meta.isIncludingFilename() && !Utils.isEmpty( meta.getFilenameField() ) ) {
+        if ( meta.isLazyConversionActive() ) {
+          outputRowData[ data.filenameFieldIndex ] = data.binaryFilename;
+        } else {
+          outputRowData[ data.filenameFieldIndex ] = data.filenames[ data.filenr - 1 ];
+        }
+      }
+
+      if ( data.isAddingRowNumber ) {
+        outputRowData[ data.rownumFieldIndex ] = data.rowNumber++;
+      }
+
+      if ( !ignoreEnclosures ) {
+        incrementLinesInput();
+      }
+
+      if ( conversionExceptions != null && conversionExceptions.size() > 0 ) {
+        // Forward the first exception
+        //
+        throw new KettleConversionException(
+          "There were " + conversionExceptions.size() + " conversion errors on line " + getLinesInput(),
+          conversionExceptions, exceptionFields, outputRowData );
+      }
+
+      return outputRowData;
+    } catch ( KettleConversionException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)